From 51691b93c5694b9b65445ec3051bec3552419545 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Fri, 2 Dec 2016 19:53:55 +0800
Subject: [PATCH] Revert "FROMLIST: PM / sleep: don't suspend parent when async
 child suspend_{noirq,late} fails"

This reverts commit 11b2dcefd88e8ae01a11c6df55d91206ac226c52.
Fixed by 04c0800c73b2 ("PM / sleep: don't suspend parent when
async child suspend_{noirq, late} fails").
---
 drivers/base/power/main.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/base/power/main.c b/drivers/base/power/main.c
index 982f586e9290..d388edadfb6a 100644
--- a/drivers/base/power/main.c
+++ b/drivers/base/power/main.c
@@ -1038,9 +1038,6 @@ static int __device_suspend_noirq(struct device *dev, pm_message_t state, bool a
 
 	dpm_wait_for_children(dev, async);
 
-	if (async_error)
-		goto Complete;
-
 	if (dev->pm_domain) {
 		info = "noirq power domain ";
 		callback = pm_noirq_op(&dev->pm_domain->ops, state);
@@ -1188,9 +1185,6 @@ static int __device_suspend_late(struct device *dev, pm_message_t state, bool as
 
 	dpm_wait_for_children(dev, async);
 
-	if (async_error)
-		goto Complete;
-
 	if (dev->pm_domain) {
 		info = "late power domain ";
 		callback = pm_late_early_op(&dev->pm_domain->ops, state);
-- 
2.35.3

